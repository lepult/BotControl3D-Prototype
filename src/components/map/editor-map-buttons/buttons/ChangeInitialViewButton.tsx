import React, { FC } from 'react';
// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
import { Button, Tooltip } from 'chayns-components';
import { useDispatch } from 'react-redux';
import clsx from 'clsx';
import { changeInitialViewState } from '../../../../redux-modules/map/actions';
import { TViewState } from '../../../../types/deckgl-map';

const ChangeInitialViewButton: FC<{
    viewState: TViewState | undefined,
    mapId: number,
}> = ({
    viewState,
    mapId,
}) => {
    const dispatch = useDispatch();

    return (
        <Tooltip
            bindListeners
            content={{ text: 'Initiale Kameraposition Ã¤ndern' }}
        >
            <Button
                className={clsx('icon-button pointer-events button--secondary')}
                onClick={() => {
                    const newInitialViewState = {
                        bearing: viewState?.bearing || 0,
                        latitude: viewState?.latitude || 0,
                        longitude: viewState?.longitude || 0,
                        pitch: viewState?.pitch || 0,
                        zoom: viewState?.zoom || 21,
                    };
                    console.log('newInitialViewState', newInitialViewState);
                    dispatch(changeInitialViewState({
                        mapId,
                        viewState: newInitialViewState,
                    }))
                }}
            >
                <i className="far fa-crosshairs-simple"/>
            </Button>
        </Tooltip>
    )
};

export default ChangeInitialViewButton;
