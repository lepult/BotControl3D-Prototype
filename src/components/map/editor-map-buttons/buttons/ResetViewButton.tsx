import React from 'react';
// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
import { Button, Tooltip } from 'chayns-components';
import clsx from 'clsx';
import { useDispatch } from 'react-redux';
import { resetViewState } from '../../../../redux-modules/misc/actions';

const ResetViewButton = () => {
    const dispatch = useDispatch();

    return (
        <Tooltip
            bindListeners
            content={{ text: 'Kameraposition zurÃ¼cksetzen' }}
        >
            <Button
                className={clsx('icon-button pointer-events button--secondary')}
                onClick={() => dispatch(resetViewState())}
            >
                <i className="far fa-location-crosshairs"/>
            </Button>
        </Tooltip>

    )
};

export default ResetViewButton;
