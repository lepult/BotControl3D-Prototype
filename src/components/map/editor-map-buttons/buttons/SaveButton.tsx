import React, { FC } from 'react';
import { createDialog, DialogButtonType, DialogType } from 'chayns-api';
import { useDispatch } from 'react-redux';
// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
import { Button } from 'chayns-components';
import clsx from 'clsx';
import { changeAdminModeType } from '../../../../redux-modules/misc/actions';
import { AdminModeType } from '../../../../types/misc';
import { ModelType } from '../../../../constants/models';

const SaveButton: FC<{
    floorModels: ModelType[],
}> = ({
    floorModels,
}) => {
    const dispatch = useDispatch();

    return (
        <Button
            className={clsx('pointer-events')}
            onClick={() => {
                console.log('Models', floorModels);
                void createDialog({
                    type: DialogType.ALERT,
                    text: "Geänderte Einstellungen werden nicht im Backend gespeichert. Mit dem Neuladen der Seite werden die Positionen deshalb zurückgesetzt.",
                    buttons: [{ type: DialogButtonType.CANCEL, text: 'Schließen' }],
                }).open().finally(() => dispatch(changeAdminModeType({ adminModeType: AdminModeType.default })));
            }}
        >
            Speichern
        </Button>
    )
};

export default SaveButton;
