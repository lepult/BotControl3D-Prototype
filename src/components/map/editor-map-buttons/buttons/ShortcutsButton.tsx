import React, { FC } from 'react';
import { createDialog, DialogButtonType, DialogResult, DialogType } from 'chayns-api';
// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
import { Button, Tooltip } from 'chayns-components';
import clsx from 'clsx';

const ShortcutsButton: FC = () => (
    <Tooltip
        bindListeners
        content={{ text: 'Shortcuts' }}
    >
        <Button
            className={clsx('icon-button pointer-events button--secondary')}
            onClick={() => {
                void createDialog({
                    type: DialogType.ALERT,
                    text: `
                        [h1]Shortcuts[/h1]
                        [p]Modell Verschieben: [b]STRG + Linke Maustaste[/b][/p]
                        [p]Modell Rotieren: [b]SHIFT + Linke Maustaste[/b][/p]
                        [p]Rückgängig: [b]STRG + Z[/b][/p]
                        [p]Wiederholen: [b]STRG + Y[/b][/p]
                    `,
                    width: '400px'
                }).open();
            }}
        >
            <i className="far fa-question"/>
        </Button>
    </Tooltip>
);

export default ShortcutsButton;
