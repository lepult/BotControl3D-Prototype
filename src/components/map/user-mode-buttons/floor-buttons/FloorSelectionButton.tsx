import React, { FC } from 'react';
import clsx from 'clsx';
// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
import { Button, Tooltip } from 'chayns-components';
import { useDispatch, useSelector } from 'react-redux';
import { selectMapById, selectSelectedMap, selectSelectedRobotId } from '../../../../redux-modules/map/selectors';
import { getPathDataByMapId } from '../../../../constants/getMapData';
import { changeSelectedMap, setFollowRobot } from '../../../../redux-modules/map/actions';
import { selectRobotStatusById } from '../../../../redux-modules/robot-status/selectors';

const FloorSelectionButton: FC<{
    mapId: number,
}> = ({
    mapId,
}) => {
    const dispatch = useDispatch();
    const selectedMap = useSelector(selectSelectedMap);
    const map = useSelector(selectMapById(mapId));
    const pathData = getPathDataByMapId(mapId);
    const selectedRobotId = useSelector(selectSelectedRobotId);
    const selectedRobot = useSelector(selectRobotStatusById(selectedRobotId || ''));

    if (map.hidden || !pathData) {
        return null;
    }

    return (
        <Tooltip
            bindListeners
            content={{ text: 'Stockwerk auswÃ¤hlen' }}
        >
            <Button
                className={clsx('list-button pointer-events', {
                    'button--secondary': selectedMap !== mapId,
                    'button--bordered': selectedRobot?.currentMap?.id === mapId && selectedMap !== mapId,
                })}
                onClick={() => {
                    dispatch(changeSelectedMap({ mapId }));
                    dispatch(setFollowRobot({ followRobot: false }));
                }}
            >
                {map.showName}
            </Button>
        </Tooltip>
    );
};

export default FloorSelectionButton;
