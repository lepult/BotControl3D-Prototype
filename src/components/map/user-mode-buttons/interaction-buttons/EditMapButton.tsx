import React from 'react';
import { useDispatch } from 'react-redux';
import clsx from 'clsx';
// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
import { Button ,Tooltip } from 'chayns-components';
import { setIsEditingMap } from '../../../../redux-modules/misc/actions';

const EditMapButton = () => {
    const dispatch = useDispatch();

    return (
        <Tooltip
            bindListeners
            content={{ text: 'Stockwerk bearbeiten' }}
        >
            <Button
                className={clsx('icon-button pointer-events button--secondary')}
                onClick={() => {
                    dispatch(setIsEditingMap(true));
                }}
            >
                <i className="far fa-pencil"/>
            </Button>
        </Tooltip>

    );
};

export default EditMapButton;
