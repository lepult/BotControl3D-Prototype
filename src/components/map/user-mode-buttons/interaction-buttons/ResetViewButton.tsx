import React from 'react';
import { useDispatch } from 'react-redux';
import clsx from 'clsx';
// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
import { Button ,Tooltip } from 'chayns-components';
import { resetViewState } from '../../../../redux-modules/misc/actions';
import { setFollowRobot } from '../../../../redux-modules/map/actions';

const ResetViewButton = () => {
    const dispatch = useDispatch();

    return (
        <Tooltip
            bindListeners
            content={{ text: 'Kameraposition zurÃ¼cksetzen' }}
        >
            <Button
                className={clsx('icon-button pointer-events button--secondary')}
                onClick={() => {
                    dispatch(resetViewState());
                    dispatch(setFollowRobot({ followRobot: false }));
                }}
            >
                <i className="far fa-location-crosshairs"/>
            </Button>
        </Tooltip>

    );
};

export default ResetViewButton;
