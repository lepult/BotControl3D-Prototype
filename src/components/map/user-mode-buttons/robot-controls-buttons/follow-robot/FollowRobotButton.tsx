import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import clsx from 'clsx';
// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
import { Button } from 'chayns-components';
import { toggleFollowRobot } from '../../../../../redux-modules/map/actions';
import { selectFollowRobot, selectSelectedRobotId } from '../../../../../redux-modules/map/selectors';

const FollowRobotButton = () => {
    const dispatch = useDispatch();

    const selectedRobotId = useSelector(selectSelectedRobotId);
    const followRobot = useSelector(selectFollowRobot);

    return (
        <Button
            disabled={!selectedRobotId}
            className={clsx('icon-button pointer-events', {
                'button--secondary': !followRobot,
            })}
            onClick={() => dispatch(toggleFollowRobot())}
        >
            <i className="far fa-cctv"/>
        </Button>
    );
};

export default FollowRobotButton;
