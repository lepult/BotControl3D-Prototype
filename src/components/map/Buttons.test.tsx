/* eslint-disable */
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, configure  } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { resetViewState, setIsEditingMap } from '../../redux-modules/misc/actions';
import EditMapButton from './user-mode-buttons/interaction-buttons/EditMapButton';
import ResetViewButton from './user-mode-buttons/interaction-buttons/ResetViewButton';
import {
    changeInitialViewState,
    changeSelectedMap,
    setFollowRobot, toggleFollowRobot,
    toggleSelectedRobot
} from '../../redux-modules/map/actions';
import ChangeInitialViewButton from './editor-map-buttons/buttons/ChangeInitialViewButton';
import { TViewState } from '../../types/deckgl-map';
import SaveButton from './editor-map-buttons/buttons/SaveButton';
import { floorModelsEg } from '../../constants/hardcoded-data/models';
import ShortcutsButton from './editor-map-buttons/buttons/ShortcutsButton';
import FloorSelectionButton from './user-mode-buttons/floor-buttons/FloorSelectionButton';
import UserModeButtons from './user-mode-buttons/UserModeButtons';
import RobotSelectionButton from './user-mode-buttons/robot-selection-buttons/RobotSelectionButton';
import FollowRobotButton from './user-mode-buttons/robot-controls-buttons/follow-robot/FollowRobotButton';

configure({ adapter: new Adapter() });

const useDispatchCallBack = jest.fn();
const openDialogCallBack = jest.fn();


const mapIdDefault = 1;
const mapIdPathData = 38;
const mapIdHidden = 1000;
const mapIdRobot = 1001;

const robotIdDefault = 'robot-default';
const robotIdSelected = 'robot-selected';
const robotIdNoPose = 'robot-no-pose';
const robotIdDefaultMap = 'robot-default-map';

const reduxStore = {
    map: {
        followRobot: false,
        selectedMap: mapIdDefault,
        selectedRobot: robotIdSelected,
        entities: {
            [mapIdDefault]: {
                id: mapIdDefault,
                name: 'Karte 1',
                showName: 'Karte 1',
                creationTime: new Date(),
            },
            [mapIdPathData]: {
                id: mapIdPathData,
                name: 'Tobit EG',
                showName: 'Tobit EG',
                creationTime: new Date(),
            },
            [mapIdHidden]: {
                id: mapIdHidden,
                name: 'Karte 2',
                showName: 'Karte 2',
                hidden: true,
                creationTime: new Date(),
            }
        },
    },
    robotStatus: {
        entities: {
            [robotIdDefault]: {
                robotStatus: {
                    robotName: 'robotDefault',
                    currentMap: {
                        id: mapIdRobot,
                    }
                },
                puduRobotStatus: {
                    robotPose: {
                        x: 10,
                        y: 10,
                        angle: 10,
                    }
                }
            },
            [robotIdSelected]: {
                robotStatus: {
                    robotName: 'robotSelected',
                    currentMap: {
                        id: mapIdRobot,
                    }
                },
                puduRobotStatus: {
                    robotPose: {
                        x: 10,
                        y: 10,
                        angle: 10,
                    }
                }
            },
            [robotIdDefaultMap]: {
                robotStatus: {
                    robotName: 'robotDefaultMap',
                    currentMap: {
                        id: mapIdDefault,
                    }
                },
                puduRobotStatus: {
                    robotPose: {
                        x: 10,
                        y: 10,
                        angle: 10,
                    }
                }
            },
            [robotIdNoPose]: {
                robotStatus: {
                    robotName: 'robotNoPose',
                    currentMap: {
                        id: mapIdRobot,
                    }
                },
                puduRobotStatus: {

                },
            }
        }
    }
}

jest.mock('react-redux', () => ({
    // @ts-ignore
    // eslint-disable-next-line @typescript-eslint/no-unsafe-return
    useDispatch: () => (e) => useDispatchCallBack(e),
    // @ts-ignore
    // eslint-disable-next-line @typescript-eslint/no-unsafe-call,@typescript-eslint/no-unsafe-return
    useSelector: (selector) => selector(reduxStore)
}));

jest.mock('chayns-components', () => {
    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
    const original = jest.requireActual('chayns-components');

    // eslint-disable-next-line @typescript-eslint/no-unsafe-return
    return {
        ...original,
        Tooltip: (props: any) => (<div className="tooltip">{props.children}</div>),
    }
});

jest.mock('chayns-api', () => {
    const original = jest.requireActual('chayns-api');

    // eslint-disable-next-line @typescript-eslint/no-unsafe-return
    return {
        ...original,
        createDialog: () => ({
            open: async () => openDialogCallBack(),
        }),
    }
});

describe('Test EditMapButton', () => {
    it('renders as expected', () => {
        const component = renderer.create(<EditMapButton/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('handles click events', () => {
        useDispatchCallBack.mock.calls = [];
        const component = shallow((<EditMapButton/>));
        component.find('Button').simulate('click');

        expect(useDispatchCallBack.mock.calls.length).toEqual(1);
        expect(useDispatchCallBack.mock.calls[0][0].type).toEqual(setIsEditingMap.type);
        expect(useDispatchCallBack.mock.calls[0][0].payload).toEqual(true);

        useDispatchCallBack.mock.calls = [];
    })
});

describe('Test ResetViewButton', () => {
    it('renders as expected', () => {
        const component = renderer.create(<ResetViewButton/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('handles click events', () => {
        useDispatchCallBack.mock.calls = [];
        const component = shallow((<ResetViewButton/>));
        component.find('Button').simulate('click');

        expect(useDispatchCallBack.mock.calls.length).toEqual(2);
        expect(useDispatchCallBack.mock.calls[0][0].type).toEqual(resetViewState.type);
        expect(useDispatchCallBack.mock.calls[0][0].payload).toEqual(undefined);
        expect(useDispatchCallBack.mock.calls[1][0].type).toEqual(setFollowRobot.type);
        expect(useDispatchCallBack.mock.calls[1][0].payload).toStrictEqual({ followRobot: false });

        useDispatchCallBack.mock.calls = [];
    })
});

describe('Test ChangeInitialViewButton', () => {
    const viewState: TViewState = {
        bearing: 10,
        latitude: 10,
        longitude: 10,
        pitch: 10,
        zoom: 10,
    };

    it('renders as expected', () => {
        const component = renderer.create(<ChangeInitialViewButton
            viewState={undefined}
            mapId={mapIdDefault}
        />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('handles click events', () => {
        useDispatchCallBack.mock.calls = [];
        const component = shallow((<ChangeInitialViewButton
            viewState={viewState}
            mapId={mapIdDefault}
        />));

        component.find('Button').simulate('click');

        expect(useDispatchCallBack.mock.calls.length).toEqual(1);
        expect(useDispatchCallBack.mock.calls[0][0].type).toEqual(changeInitialViewState.type);
        expect(useDispatchCallBack.mock.calls[0][0].payload).toStrictEqual({
            mapId: mapIdDefault,
            viewState,
        });

        useDispatchCallBack.mock.calls = [];
    })
});
describe('Test ResetViewButton', () => {
    it('renders as expected', () => {
        const component = renderer.create(<SaveButton floorModels={[]}/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('handles click events', async () => {
        useDispatchCallBack.mock.calls = [];
        const component = shallow((<SaveButton floorModels={[]}/>));
        component.find('Button').simulate('click');

        await new Promise(r => setTimeout(r, 100));

        expect(useDispatchCallBack.mock.calls.length).toEqual(1);
        expect(useDispatchCallBack.mock.calls[0][0].type).toEqual(setIsEditingMap.type);
        expect(useDispatchCallBack.mock.calls[0][0].payload).toEqual(false);

        useDispatchCallBack.mock.calls = [];
    })
});

describe('Test ShortcutsButton', () => {
    it('renders as expected', () => {
        const component = renderer.create(<ShortcutsButton/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('handles click events', async () => {
        openDialogCallBack.mock.calls = [];
        const component = shallow((<ShortcutsButton/>));
        component.find('Button').simulate('click');

        await new Promise(r => setTimeout(r, 100));

        expect(openDialogCallBack.mock.calls.length).toEqual(1);

        openDialogCallBack.mock.calls = [];
    })
});

describe('Test FloorSelectionButton', () => {
    it('renders as expected', () => {
        reduxStore.map.selectedMap = mapIdDefault;

        // Map is not selected
        const component = renderer.create(<FloorSelectionButton mapId={mapIdPathData}/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        // Map is selected
        component.update(<FloorSelectionButton mapId={mapIdDefault}/>);
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        reduxStore.map.selectedMap = mapIdDefault;
    });

    it('shows border when map has selected robot', () => {
        reduxStore.map.selectedMap = mapIdDefault;

        // Map is not selected and has selected Robot
        const component = renderer.create(<FloorSelectionButton mapId={mapIdRobot}/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        // Map is selected and has selected Robot
        reduxStore.map.selectedMap = mapIdRobot;
        component.update(<FloorSelectionButton mapId={mapIdRobot}/>);
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        reduxStore.map.selectedMap = mapIdDefault;
    });

    it('doesnt render when map is hidden or doesnt have pathData', () => {
        // Map is hidden
        const component = renderer.create(<FloorSelectionButton mapId={mapIdHidden}/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        // Map has no pathData
        component.update(<FloorSelectionButton mapId={mapIdDefault}/>);
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('handles click events', async () => {
        useDispatchCallBack.mock.calls = [];
        const component = shallow((<FloorSelectionButton mapId={mapIdPathData}/>));
        console.log('component', component.debug());
        component.find('Button').simulate('click');

        expect(useDispatchCallBack.mock.calls.length).toEqual(2);
        expect(useDispatchCallBack.mock.calls[0][0].type).toEqual(changeSelectedMap.type);
        expect(useDispatchCallBack.mock.calls[0][0].payload).toStrictEqual({ mapId: mapIdPathData });
        expect(useDispatchCallBack.mock.calls[1][0].type).toEqual(setFollowRobot.type);
        expect(useDispatchCallBack.mock.calls[1][0].payload).toStrictEqual({ followRobot: false });

        useDispatchCallBack.mock.calls = [];
    })
});


describe('Test RobotSelectionButton', () => {
    it('renders as expected', () => {
        // Robot is not selected
        const component = renderer.create(<RobotSelectionButton robotId={robotIdDefault}/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        // Robot is selected
        component.update(<RobotSelectionButton robotId={robotIdSelected}/>);
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('doesnt render when robot has no pose', () => {
        // Robot is not selected
        const component = renderer.create(<RobotSelectionButton robotId={robotIdNoPose}/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        // Robot is selected
        component.update(<RobotSelectionButton robotId={robotIdSelected}/>);
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('handles click events', async () => {
        useDispatchCallBack.mock.calls = [];
        const component = shallow((<RobotSelectionButton robotId={robotIdDefault}/>));
        component.find('Button').simulate('click');

        expect(useDispatchCallBack.mock.calls.length).toEqual(2);
        expect(useDispatchCallBack.mock.calls[0][0].type).toEqual(toggleSelectedRobot.type);
        expect(useDispatchCallBack.mock.calls[0][0].payload).toStrictEqual({ robotId: robotIdDefault });
        expect(useDispatchCallBack.mock.calls[1][0].type).toEqual(changeSelectedMap.type);
        expect(useDispatchCallBack.mock.calls[1][0].payload).toStrictEqual({ mapId: reduxStore.robotStatus.entities[robotIdDefault].robotStatus.currentMap.id });

        useDispatchCallBack.mock.calls = [];
    });
});

describe('Test FollowRobotButton', () => {
    it('uses correct colors when following and not following', () => {
        // Not following robot
        reduxStore.map.followRobot = false;
        const component = renderer.create(<FollowRobotButton/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        // Following robot
        reduxStore.map.followRobot = true;
        component.update(<FollowRobotButton/>);
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        reduxStore.map.followRobot = false;
    });

    it('disabled when no robot is selected', () => {
        // Robot selected
        reduxStore.map.selectedRobot = robotIdSelected;
        const component = renderer.create(<FollowRobotButton/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        // No robot selected
        reduxStore.map.selectedRobot = '';
        component.update(<FollowRobotButton/>);
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        reduxStore.map.selectedRobot = robotIdSelected;
    });

    it('handles click events', async () => {
        useDispatchCallBack.mock.calls = [];
        const component = shallow((<FollowRobotButton/>));
        component.find('Button').simulate('click');

        expect(useDispatchCallBack.mock.calls.length).toEqual(2);
        expect(useDispatchCallBack.mock.calls[0][0].type).toEqual(toggleFollowRobot.type);

        useDispatchCallBack.mock.calls = [];
    });
});
