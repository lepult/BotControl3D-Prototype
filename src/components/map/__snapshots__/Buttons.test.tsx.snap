// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ChangeInitialViewButton renders as expected 1`] = `
<div
  className="tooltip"
>
  <button
    className="icon-button pointer-events button--secondary button"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <i
      className="far fa-crosshairs-simple"
    />
  </button>
</div>
`;

exports[`Test EditMapButton renders as expected 1`] = `
<div
  className="tooltip"
>
  <button
    className="icon-button pointer-events button--secondary button"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <i
      className="fa-duotone fa-layer-group"
    />
  </button>
</div>
`;

exports[`Test FloorSelectionButton doesnt render when map is hidden or doesnt have pathData 1`] = `null`;

exports[`Test FloorSelectionButton doesnt render when map is hidden or doesnt have pathData 2`] = `null`;

exports[`Test FloorSelectionButton renders as expected 1`] = `
<div
  className="tooltip"
>
  <button
    className="list-button pointer-events button--secondary button"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Tobit EG
  </button>
</div>
`;

exports[`Test FloorSelectionButton renders as expected 2`] = `null`;

exports[`Test FloorSelectionButton shows border when map has selected robot 1`] = `null`;

exports[`Test FloorSelectionButton shows border when map has selected robot 2`] = `null`;

exports[`Test FollowRobotButton disabled when no robot is selected 1`] = `
<div
  className="tooltip"
>
  <button
    className="icon-button pointer-events button--secondary button"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <i
      className="far fa-cctv"
    />
  </button>
</div>
`;

exports[`Test FollowRobotButton disabled when no robot is selected 2`] = `
<div
  className="tooltip"
>
  <button
    className="icon-button pointer-events button--secondary button button--disabled"
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    <i
      className="far fa-cctv"
    />
  </button>
</div>
`;

exports[`Test FollowRobotButton uses correct colors when following and not following 1`] = `
<div
  className="tooltip"
>
  <button
    className="icon-button pointer-events button--secondary button"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <i
      className="far fa-cctv"
    />
  </button>
</div>
`;

exports[`Test FollowRobotButton uses correct colors when following and not following 2`] = `
<div
  className="tooltip"
>
  <button
    className="icon-button pointer-events button"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <i
      className="far fa-cctv"
    />
  </button>
</div>
`;

exports[`Test ResetViewButton renders as expected 1`] = `
<div
  className="tooltip"
>
  <button
    className="icon-button pointer-events button--secondary button"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <i
      className="far fa-location-crosshairs"
    />
  </button>
</div>
`;

exports[`Test ResetViewButton renders as expected 2`] = `
<button
  className="pointer-events button"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  Speichern
</button>
`;

exports[`Test RobotSelectionButton doesnt render when robot has no pose 1`] = `null`;

exports[`Test RobotSelectionButton doesnt render when robot has no pose 2`] = `
<div
  className="tooltip"
>
  <button
    className="pointer-events robot-selection-button button"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    robotSelected
  </button>
</div>
`;

exports[`Test RobotSelectionButton renders as expected 1`] = `
<div
  className="tooltip"
>
  <button
    className="pointer-events robot-selection-button button--secondary button"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    robotDefault
  </button>
</div>
`;

exports[`Test RobotSelectionButton renders as expected 2`] = `
<div
  className="tooltip"
>
  <button
    className="pointer-events robot-selection-button button"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    robotSelected
  </button>
</div>
`;

exports[`Test ShortcutsButton renders as expected 1`] = `
<div
  className="tooltip"
>
  <button
    className="icon-button pointer-events button--secondary button"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <i
      className="far fa-question"
    />
  </button>
</div>
`;
